DockerReadme.md

```diff
@@ -0,0 +1,77 @@
+# Configuring the docker environment.
+
+### For Linux you can install docker directly, but for mac and windows a linux vm is required to host the docker containers.
+
+Installation instructions for mac and windows can be found;
+
+here
+
+https://docs.docker.com/installation/mac/
+
+and here
+
+https://docs.docker.com/installation/windows/
+
+### Building the docker container. (This assumes boot2docker is running and working as per the instructions above if you are
+on mac or windows.)
+
+#### background
+Docker requires a container to be built in order for it to run our code. Our current scenario downloads a base image
+that is listed in the Dockerfile. That image contains installations of node, gulp and bower. Once that images is
+downloaded it then runs apt-get to install updates to the packages so we are using the latest versions. (This may not be
+the desired behavior at some point, but during development it is how things are working.)
+
+Once the container has been updated to use the newest versions of the software we need, it will run through the build
+process using npm install and gulp, and when that is done the process is complete.
+
+Note: This seems to have issues on slower network connections, and can take a few minutes to happen.
+
+#### building
+###### Note: Make sure the working directory is cleaned of the node_modules or there could be build errors because docker will copy the path supplied to the build command into the container, which will include node_modules if it has not been removed.
+
+run the build command to create the docker container. The -t flag sets a tag for the container created, and the first argument to build tells it what path to look for the Dockerfile and files to copy into the container. When running a docker build from within the permisive directory run the following command.
+
+`docker build -t="atsid/permissive" .`
+
+#### Setting up the environment variables.
+
+At this time the permissive configuration values will be set with an environment file when running from docker. In order
+to do this a file with the following environment variables is needed.
+
+GITHUB_ORG=<org name of repos to be managed.>
```

Comment:
you may instead want to just refer back to the environment variables in the main readme, noting that the properties file needs to have a "valid set", which could include the github oauth params, or even just the mock params.